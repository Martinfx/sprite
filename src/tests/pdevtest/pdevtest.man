'\" Copyright 1989 Regents of the University of California
'\" Permission to use, copy, modify, and distribute this
'\" documentation for any purpose and without fee is hereby
'\" granted, provided that this notice appears in all copies.
'\" The University of California makes no representations about
'\" the suitability of this material for any purpose.  It is
'\" provided "as is" without express or implied warranty.
'\" 
'\" $Header: /sprite/lib/forms/RCS/proto.man,v 1.5 89/01/27 08:36:02 ouster Exp $ SPRITE (Berkeley)
'/" 
.so \*(]ltmac.sprite
.HS PDEVTEST cmds
.BS
.SH NAME
pdevtest \- Test program for pseudo-devices
.SH SYNOPSIS
.nf
\fBpdevtest\fR [\fIoptions\fR]
.SH OPTIONS
.IP "\fB\-S\fR" 14
Run as a slave process.
.IP "\fB\-M \fInumSlaves\fR" 14
Run as the master of \fInumSlaves\fR.
.IP "\fB\-o \fIoutFile\fR" 14
Name of the output file.
.IP "\fB\-p \fIpseudoDevice\fR" 14
Name of the pseudo-device.
.IP "\fB\-i\fR" 14
Echo data to/from the pseudo-device using ioctl.
.IP "\fB\-r\fR" 14
Test reading from a pseudo-device.
.IP "\fB\-w\fR" 14
Test writing to a pseudo-device.
.IP "\fB\-W\fR" 14
Enable write-behind (use with \fB-M\fP).
.IP "\fB\-s\fR" 14
Use select, too.
.IP "\fB\-c\fR" 14
Fork the client process.
.IP "\fB\-n \fInumReps\fR" 14
Number of test repetitions.
.IP "\fB\-d \fInumBytes\fR" 14
Number of bytes to transfer.
.IP "\fB\-P \fInumLoops\fR" 14
Loop \fInumLoops\fP times before replying (with \fB-M\fP).
.IP "\fB\-z\fR" 14
Force Master to use select with 1 client.
.IP "\fB\-b\fR" 14
Test switching request buffers (use with \fB-M\fP).
.IP "\fB\-x\fR" 14
Turn off all tracing.
.IP "\fB\-h\fR" 14
Except leave histogram taking on (ok with \fB-x\fP).
.BE

.SH DESCRIPTION
.PP
\fBpdevtest\fP is used to test various features of pseudo-devices.
Several instances of the program are run for a single test.
One master process (\fB-M\fP) is run first;
it is the pseudo-device server.
The master can enable write-behind (\fB-W\fP),
pause before replying (\fB-P\fP),
and switch request buffers (\fB-b\fP).
Ordinarily if there is only one slave a slightly faster service
loop is used to does not use select.
The master process can be forced to use select even in this case with \fB-z\fP.
After the master has started several slave (\fB-S\fP) processes are started.
Each slave can do a different test (\fB-i -r -w\fP).
The number of repetitions and the amount of data to transfer
are specified with the \fB-n\fP and \fB-d\fP options.
Additionally, a slave process can fork (\fB-c\fP) to test stream sharing,
and use select (\fB-s\fP).
.pp
\fBpdevtest\fP generates output to the file specified by the \fB-o\fP option.
It can also use a specific pseudo-device file (\fB-p\fP).
Finally, system tracing can be turned off (\fB-x\fP),
which makes the tests run faster.

.SH KEYWORDS
pseudo-device, test

