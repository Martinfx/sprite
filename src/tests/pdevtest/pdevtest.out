pdevtest 
S: Idle ticks 0/170160 =   0.00% Idle, Context Sw. 504 inv 139 full 424, Elapsed time 9.450
Cache blocks max 3622 min 32 number 1860/1860 free 59/59 limit 3622
Bytes read 0+0 remote 0 disk 0+0
Bytes written 374+0 remote 0 disk 0+0	traffic ratio %0
Dev bytes read 2184 written 5144
Zero Fills read 0 write1 0 write2 0 frag 0
Appends 0 Overwrites 0 Failed Reads 0
Cache reads 0 hits: dirty 0 clean 0 other 0
Cache writes 2 hits 2 misses 0 thru 0	traffic ratio %0
Write thru 0 data 0 indirect 0 desc 0 dir 0
VM requests 0 tried 0 gave 0
Cache blocks created 0, alloc from free 0 part 0 lru 0
Handles 720 created 1 installed 2 hits 1 old 0 version 0 flush 0
	fetched 3 hits 1 released 27 locks 1302/1302 wait 0
Segments fetched 1 hits 1
Lookup relative 2 absolute 1 redirect 0 found 0 loops 0 timeouts 0 stale 0
Kernel VM Pages: 285 (Code+Data=255 Stacks=30)
User VM Pages:   826 (Free=192 Dirty=0 Res=3 Alloc-list=631)
Modified pages: Total=141 %Tot-dirty=17.13 %Inuse-dirty=22.35
FS Pages: Current=930 Max=1554 Min=0
Faults:       29 (Zero=11 FS=2 Swap=0 Quick=11 Coll=0)
              29 (Code=0 Heap=18 Stack=11)
Mod page stats:  Pot-mod=0 Not-mod=0 Not-hard-mod=0
COW: Heap (1/10)=10% Stk (0/3)=0% Tot (1/13)=7%
     Quick (0/1)=0%
COR: Heap (8/10)=80% Stk (1/3)=33% Tot (9/13)=69%
COR-mod: Heap(4/8)=50% Stk (1/1)=100% Tot (5/9)=55%
Swap pages copied: 0
Vm allocs: 24 (Free=24 From-FS=0 From-alloc-list=0)
VM-FS stats: Asked=0 Free-pages=0 Allocs=0 Frees=0
Alloc-list searches: 0 (Free=0 In-use=0)
Extra-searches: 0 (Lock=0 Ref=0 Dirty=0)
Pages written 0
